src/branchandbound.jl

```diff
@@ -68,3 +68,27 @@ function divide_dom!(p::Union{TaylorN{T}, Taylor1{T}},
     R = append!(RR, R[(index + 1):length(R)])
     return D, R
 end
+
+function enclose_binary(f, dom::Interval; kmax=3, tol=1e-3, algorithm=:IntervalArithmetic)
+    y = enclose(f, dom, algorithm=algorithm)
+    yinf, ysup = inf(y), sup(y)
+    x = bisect(dom)
+    fx1 = enclose(f, x[1], algorithm=algorithm)
+    fx2 = enclose(f, x[2], algorithm=algorithm)
+    ynew = union(fx1, fx2)
+    ynew_inf, ynew_sup = inf(ynew), sup(ynew)
+    no_cambia_inf = abs(yinf - ynew_inf) < tol
+    no_cambia_sup = abs(ysup - ynew_sup) < tol
+    no_cambia_ninguno = (no_cambia_inf && no_cambia_sup)
+    empeora_inf = ynew_inf < yinf
+    empeora_sup = ynew_sup > ysup
+    empeoran_ambos = empeora_inf && empeora_sup
+    if kmax == 0 || no_cambia_ninguno || empeoran_ambos
+        return Interval(yinf, ysup)
```

Comment:
I would move the base condition for the recursion just after line 74. In cases like this we can do a short-circuit and, like so:

```julia
k == 0 && return Interval(yinf, ysup)
```